clang-tools-extra/clang-tidy/bugprone/TaggedUnionMemberCountCheck.cpp

```diff
@@ -153,61 +178,57 @@ size_t TaggedUnionMemberCountCheck::getNumberOfValidEnumValues(
         CeIsLast = true;
         CeValue = Val;
         CeCount += 1;
-        CountingEnumConstantDecl = Enumerator;
+        OutCountingEnumConstantDecl = Enumerator;
       } else {
         CeIsLast = false;
       }
     }
   }
 
-  size_t ValidValuesCount = EnumValues.size();
+  std::size_t ValidValuesCount = EnumValues.size();
   if (CeCount == 1 && CeIsLast && CeValue == MaxTagValue) {
     ValidValuesCount -= 1;
   } else {
-    CountingEnumConstantDecl = nullptr;
+    OutCountingEnumConstantDecl = nullptr;
   }
 
   return ValidValuesCount;
 }
 
 void TaggedUnionMemberCountCheck::check(
     const MatchFinder::MatchResult &Result) {
-  const auto *Root = Result.Nodes.getNodeAs<RecordDecl>("root");
-  const auto *UnionField = Result.Nodes.getNodeAs<FieldDecl>("union");
-  const auto *TagField = Result.Nodes.getNodeAs<FieldDecl>("tags");
-
-  // The matcher can only narrow down the type to recordType()
-  if (!isUnion(UnionField))
-    return;
-
-  if (hasMultipleUnionsOrEnums(Root))
-    return;
+  const auto *Root = Result.Nodes.getNodeAs<RecordDecl>(RootMatchBindName);
+  const auto *UnionField =
+      Result.Nodes.getNodeAs<FieldDecl>(UnionMatchBindName);
+  const auto *TagField = Result.Nodes.getNodeAs<FieldDecl>(TagMatchBindName);
+  assert(Root && UnionField && TagField);
```

Comment:
How about adding a return if one of these is missing, so we don't crash in release mode? Also it's a good idea to separate the assertions, so that we know which of the 3 conditions fails on a crash.